### ISSUE TITLE
Unable to sign file with the example code from README

### ISSUE THREAD

**Issue by contrun** [ANSSI-FR/libecc#30]
I tried to run the following code (adapted from README, used variables to make sure there are no typos).

```
export curve=SECP256R1 sign_algo=ECDSA hash_algo=SHA3_512 file=myfile keyname=mykeypair
echo test > "$file"; ./build/ec_utils gen_keys "$curve" "$sign_algo" "$keyname"; ./build/ec_utils sign $curve $sign_algo SHA3_512 "$file" "${keyname}_private_key.bin" sig.bin
```

The above commands output `sign error ...`.

I am using gcc 11.2.0 on NixOS 22.05. Please let me know if there are any additional information I need to provide.

**Comment by rb-anssi** [comment]
Hi,

Thanks for your interest in libecc and for the confirmed issue. This should be resolved in commit 5f43dcc319926156e851c9452454da466d8173f3.

Regards,

**Comment by contrun** [comment]
Thanks for prompt reply.
I tried newest code. I now works. Thansk.
